File: LeetCode_Marathons/LCM3/Easy/Longest_Even_Odd_Subarray_With_Threshold/a.java
class Solution {
    public int longestAlternatingSubarray(int[] nums, int threshold) {
        int n = nums.length;
        for (int arrSz = n; arrSz >= 1; arrSz--) {
            for (int startArr = 0; startArr <= n - arrSz; startArr++) {
                Boolean isCorrect = true;
                if (nums[startArr] % 2 != 0) {
                    isCorrect = false;
                    continue;
                }
                for (int i = startArr; i <= arrSz + startArr - 2; i++) {
                    if (nums[i] % 2 == nums[i + 1] % 2 || nums[i] > threshold) {
                        isCorrect = false;
                        break;
                    }
                }
                if (nums[arrSz + startArr - 1] > threshold) {
                    isCorrect = false;
                }
                if (isCorrect) return arrSz;
            }
        }
        return 0;
    }
}

File: LeetCode_Marathons/LCM3/Easy/Longest_Even_Odd_Subarray_With_Threshold/a.py
class Solution:
    def longestAlternatingSubarray(self, nums: List[int], threshold: int) -> int:
        n: int = len(nums)
        for arrSz in range(n, 0, -1):
            for startArr in range(0, n - arrSz + 1):
                isCorrect: bool = True
                if nums[startArr] % 2 != 0:
                    isCorrect = False
                    continue
                for i in range(startArr, arrSz + startArr - 1):
                    if nums[i] % 2 == nums[i + 1] % 2 or nums[i] > threshold:
                        isCorrect = False
                        break
                if nums[arrSz + startArr - 1] > threshold: isCorrect = False
                if isCorrect: return arrSz
        return 0